prov/util/src/util_mem_hooks.c

```diff
@@ -540,7 +587,10 @@ static bool ofi_memhooks_valid(struct ofi_mem_monitor *monitor,
 
 static int ofi_memhooks_start(struct ofi_mem_monitor *monitor)
 {
-	int i, ret;
+	int i, ret, i_err;
+	static bool first_time = true;
```

Comment:
I don't think you need the first_time variable.  The subscribe check on line 595 will protect from Libfabric patching itself.  If someone else patched, we can just keep trying and rolling back, because that's not the common case.